src/org/labkey/test/components/domain/DomainFieldRow.java

```diff
@@ -492,6 +500,54 @@ public boolean hasFieldWarning()
         return getComponentElement().getAttribute("class").contains("domain-row-border-warning");
     }
 
+    // conditional formatting and validation options
+
+    public RangeValidatorDialog clickAddRange()
+    {
+        expand();
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().addRangeButton()));
+        elementCache().addRangeButton().click();
+        return new RangeValidatorDialog(this, getDriver());
+    }
+    public RangeValidatorDialog clickEditRanges()
+    {
+        expand();
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().editRangesButton()));
+        elementCache().editRangesButton().click();
+        return new RangeValidatorDialog(this, getDriver());
+    }
+
+    public RegexValidatorDialog clickAddRegex()
+    {
+        expand();
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().addRegexValidatorButton()));
+        elementCache().addRegexValidatorButton().click();
+        return new RegexValidatorDialog(this, getDriver());
+    }
+    public RegexValidatorDialog clickEditRegex()
+    {
+        expand();
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().editRegexValidatorButton()));
+        elementCache().editRegexValidatorButton().click();
+        return new RegexValidatorDialog(this, getDriver());
+    }
+
+    public ConditionalFormatDialog clickAddFormat()
+    {
+        expand();
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().addFormatButton()));
+        elementCache().addFormatButton().click();
+        return new ConditionalFormatDialog(this, getDriver());
+    }
+
+    public ConditionalFormatDialog clickEditFormats()
+    {
+        expand();
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().editFormatButton()));
+        elementCache().editFormatButton().click();
+        return new ConditionalFormatDialog(this, getDriver());
+    }
+
```

Comment:
Yes, that will be nice.  I'm implementing this pattern for RegexValidator, which is easy.
RangeValidator runs into complications converting FieldDefinition.RangeType to Filter.Operator and back.

Because I'm using Filter.Operator.toString() to select operators by value, (and its output equals exactly the url params) I'm weighing the ups and downs of using either one